modules/programs/firefox.nix

```diff
@@ -99,17 +190,34 @@ in
       in
         [ (wrapper cfg.package { }) ];
 
-    home.file.".mozilla/${extensionPath}" = mkIf (cfg.extensions != []) (
-      let
-        extensionsEnv = pkgs.buildEnv {
-          name = "hm-firefox-extensions";
-          paths = cfg.extensions;
-        };
-      in
-        {
-          source = "${extensionsEnv}/share/mozilla/${extensionPath}";
-          recursive = true;
-        }
-    );
+    home.file = mkMerge ([{
+      ".mozilla/${extensionPath}" = mkIf (cfg.extensions != []) (
+        let
+          extensionsEnv = pkgs.buildEnv {
+            name = "hm-firefox-extensions";
+            paths = cfg.extensions;
+          };
+        in
+          {
+            source = "${extensionsEnv}/share/mozilla/${extensionPath}";
+            recursive = true;
+          }
+      );
+
+      ".mozilla/firefox/profiles.ini" = mkIf (cfg.profiles != {}) {
+        text = profilesINI;
+        recursive = true;
+      };
+    }] ++ (mapAttrsToList (_: profile: {
```

Comment:
A bit nicer, I think, to have
```
flip mapAttrsToList cfg.profiles (_: profile: {
  …
})
```

Reply:
Changed, thx, didn't knew about flip function